sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -124,10 +126,75 @@ class RapidsExecutorPlugin extends ExecutorPlugin with Logging {
       extraConf: util.Map[String, String]): Unit = {
     try {
       val conf = new RapidsConf(extraConf.asScala.toMap)
+      val cudfPropertiesFileName = "cudf-java-version-info.properties"
+      val pluginPropertiesFileName = "rapids4spark-version-info.properties"
       if (conf.shimsProviderOverride.isDefined) {
         ShimLoader.setSparkShimProviderClass(conf.shimsProviderOverride.get)
       }
 
+      // Compare if the cudf version mentioned in the classpath is equal to the version which plugin
+      // expects. If there is a version mismatch, throw error. This check can be disabled by
+      // setting this config spark.rapids.cudfVersionOverride=true
+      val props = new Properties
+      val cudfClassLoader = classOf[ai.rapids.cudf.ColumnVector].getClassLoader
+      val cudfProperties = cudfClassLoader.getResourceAsStream(cudfPropertiesFileName)
+      if (cudfProperties == null) {
+        val errorMsg = s"Could not find properties file $cudfPropertiesFileName in " +
+          "the cudf jar. Cannot verify cudf version compatibility with RAPIDS Accelerator version."
+        if (!conf.cudfVersionOverride) {
+          throw new RuntimeException(errorMsg)
+        } else {
+          logWarning(errorMsg)
```

Comment:
Note that this logs and continues, leading to an NPE exception below.  Similar issues with other places that log a warning here.  Probably cleaner to create a specific exception type, throw that exception whenever there's an issue, then have the calling code catch the exception and log the exception message if the override is set rather than make it fatal.  Otherwise we have to put early returns in all these places where we're logging.

Reply:
Yes, you are right. Will update the PR to use custom exception wherever we are logging as suggested. Thanks!